app/src/main/java/io/appium/settings/RecorderService.java

```diff
@@ -0,0 +1,135 @@
+package io.appium.settings;
+
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.media.projection.MediaProjection;
+import android.media.projection.MediaProjectionManager;
+import android.os.Build;
+import android.os.IBinder;
+import android.util.DisplayMetrics;
+import android.util.Log;
+
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+import io.appium.settings.helpers.NotificationHelpers;
+
+public class RecorderService extends Service {
+    private static final String TAG = "RecorderService";
+
+    public static final int REQUEST_CODE_SCREEN_CAPTURE = 123;
+    private static final String BASE = "io.appium.settings.recording";
+    public static final String ACTION_RECORDING_START = BASE + ".ACTION_START";
+    public static final String ACTION_RECORDING_STOP = BASE + ".ACTION_STOP";
+    public static final String ACTION_RECORDING_RESULT_CODE = "result_code";
+    public static final String ACTION_RECORDING_FILENAME = "recording_filename";
```

Comment:
just `path` would be file

Reply:
since users will only be allowed to type custom filename of recording file (we dont allow custom paths we allow only custom filename) i thought "filename" is more appropriate but again i can change it if you want  👍